python/lsst/pipe/tasks/dcrAssembleCoadd.py

```diff
@@ -0,0 +1,781 @@
+        default=False,
+    )
+    maxGain = pexConfig.Field(
+        dtype=float,
+        doc="Maximum convergence-weighted gain to apply between forward modeling iterations.",
+        default=2.,
+    )
+    minGain = pexConfig.Field(
+        dtype=float,
+        doc="Minimum convergence-weighted gain to apply between forward modeling iterations.",
+        default=0.5,
+    )
+    convergenceMaskPlanes = pexConfig.ListField(
+        dtype=str,
+        default=["DETECTED"],
+        doc="Mask planes to use to calculate convergence."
+    )
+    lambdaEff = pexConfig.Field(
+        dtype=float,
+        doc="Effective center wavelength of the filter, in nm. This will be replaced in DM-13668.",
+        default=478.,
+    )
+    filterWidth = pexConfig.Field(
+        dtype=float,
+        doc="Width of the filter, in nm. This will be replaced in DM-13668",
+        default=147.,
+    )
+
+    def setDefaults(self):
+        CompareWarpAssembleCoaddConfig.setDefaults(self)
+        self.assembleStaticSkyModel.removeMaskPlanes = []
+        self.assembleStaticSkyModel.warpType = 'direct'
+        self.removeMaskPlanes = []
+        self.warpType = 'direct'
+        self.statistic = 'MEAN'
+        if self.usePsf:
+            self.useFFT = True
+        if self.doWeightGain:
+            self.useConvergence = True
+
+
+class DcrAssembleCoaddTask(CompareWarpAssembleCoaddTask):
+
+    ConfigClass = DcrAssembleCoaddConfig
+    _DefaultName = "dcrAssembleCoadd"
+
+    @pipeBase.timeMethod
+    def run(self, dataRef, selectDataList=[]):
+        """! Assemble a coadd from a set of Warps.
+
+        Coadd a set of Warps. Compute weights to be applied to each Warp and find scalings to
+        match the photometric zeropoint to a reference Warp. Optionally, match backgrounds across
+        Warps if the background has not already been removed. Assemble the Warps using
+        \ref assemble. Interpolate over NaNs and optionally write the coadd to disk. Return the coadded
+        exposure.
+
+        \anchor runParams
+        @param[in] dataRef: Data reference defining the patch for coaddition and the reference Warp
+                        (if config.autoReference=False). Used to access the following data products:
+                        - [in] self.config.coaddName + "Coadd_skyMap"
+                        - [in] self.config.coaddName + "Coadd_ + <warpType> + "Warp" (optionally)
+                        - [out] self.config.coaddName + "Coadd"
+        @param[in] selectDataList[in]: List of data references to Warps. Data to be coadded will be
+                                   selected from this list based on overlap with the patch defined by dataRef.
+
+        @return a pipeBase.Struct with fields:
+                 - coaddExposure: coadded exposure
+                 - nImage: exposure count image
+        """
+        retStruct = AssembleCoaddTask.run(self, dataRef, selectDataList=selectDataList)
```

Comment:
I usually like to spell this `results`, which avoids the awkward "ret" prefix and the Hungarian postfix.